### PR TITLE
MOLO-9134 docs for api endpoint for mojo

### PR DESCRIPTION
**Pull request by danyilmaz**
Added docs for the proposed api for mojo

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -57,49 +52,191 @@ code:<AUTH_CODE>
...
+    - "other"
+- `first_name` The customer's first name
+- `middle_name` The customer's middle name
+- `last_name` The customer's last name
+- `date_of_birth` The customer's date of birth, in ISO 8601 format (yyyy-mm-dd)
+- `mobile` The customer's mobile number in the form "+441615676878"
+- `address` An address object representing the customer's address, in the form:
+    - `address_type` The customer's address type, a choice of:
+        - "current"
+        - "previous"
```
Comment (sonerayberk) [inline source/index.html.md]: Are you sure that 2 addresses type are available? For getting a DIP we require to have at least `current` address type

Comment (sonerayberk) [reply source/index.html.md]: If `address` is not a list, but a single object - we require to have `current` address type

Comment (danyilmaz) [reply source/index.html.md]: I thought they needed to give us their address history for a period of ?3 years or something?
I'm imagining this as a list which we'll do some validation on

File: source/index.html.md [hunk 2]
```diff
@@ -57,49 +52,191 @@ code:<AUTH_CODE>
...
-curl -X GET \
-  https://partner.molofinance.com/v1/applications/ \
-  -H 'Authorization: Bearer <ACCESS TOKEN>' \
-  -H 'cache-control: no-cache'
+curl
+```
+**Send a `POST` request to https://partner.molofinance.com/v1/applications/ to create a new application.**
+You will need to provide the following parameters:
+
+- `property_value` A decimal value with 2 decimal places. This value is presumed to be in GBP
```
Comment (SergeyKubrak) [inline source/index.html.md]: Can we define field's max length.

File: source/index.html.md (hunk 1 above)
Comment (SergeyKubrak) [reply source/index.html.md]: For Dip current residential address enough. FO full application yes, we request custoemr to provide resedential address history, but it's more for internal LP.

**Review by SergeyKubrak — CHANGES_REQUESTED** [review]

File: source/index.html.md (hunk 2 above)
Comment (danyilmaz) [reply source/index.html.md]: Sure, do we just want that on the decimal/money fields, or do we need that on the other fields too?

File: source/index.html.md (hunk 1 above)
Comment (danyilmaz) [reply source/index.html.md]: I'll just make this endpoint accept a single address object and have it default to `current` then

File: source/index.html.md [hunk 3]
```diff
@@ -136,54 +144,549 @@ https://oauth.molofinance.com/token/
...
+{
+    "decision": "valid"
+}
+```
+
+**Send a `POST` request to our applications dip endpoint to get a decision in principal.**
+
+[https://partner.molofinance.com/api/v1/applications/{pk}/get-dip/](https://partner.molofinance.com/api/v1/applications/{pk}/get-dip/)
+
+You must send a the parameter:
```
Comment (sonerayberk) [inline source/index.html.md]: typo

File: source/index.html.md [hunk 4]
```diff
@@ -136,54 +144,549 @@ https://oauth.molofinance.com/token/
...
+
+- `dip_consent` A boolean, you must send "true" to get a dip
+
+The response will be in the form:
+
+- `dip_agreed` Whether Molo can give you a decision in principal. A boolean.
+- `link_to_dip` A link to the DIP itself so that it can be downloaded.
```
Comment (sonerayberk) [inline source/index.html.md:363]: I think it's outdated

File: source/index.html.md [hunk 5]
```diff
@@ -136,54 +144,549 @@ https://oauth.molofinance.com/token/
...
+```shell
+curl -X POST \
+    http://partner.molofinance.com/api/v1/applications/<APPLICATION_ID>/get-dip/ \
+    -H 'Accept: */*' \
+    -H 'Authorization: Bearer <TOKEN>' \
+    -H 'Content-Type: application/json' \
+    -H 'Host: partner.molofinance.com' \
+    -d '{' \
+            '"dip_consent": "true"', \
+       '}'
```
Comment (sonerayberk) [inline source/index.html.md:341]: Shouldn't there be other consents?

File: source/index.html.md [hunk 6]
```diff
@@ -136,54 +144,549 @@ https://oauth.molofinance.com/token/
...
+
+```shell
+curl -X DELETE \
+    http://partner.molofinance.com/api/v1/applicants/<APPLICANT_ID> \
+    -H 'Accept: application/json' \
+    -H 'Authorization: Bearer <TOKEN>' \
+    -H 'Content-Type: application/json' \
+```
+
+> **Expected response: Status 200**
```
Comment (sonerayberk) [inline source/index.html.md]: There will be another status: `HTTP 204 NO CONTENT`

File: source/index.html.md (hunk 5 above)
Comment (danyilmaz) [reply source/index.html.md:341]: not here, no

File: source/index.html.md (hunk 4 above)
Comment (danyilmaz) [reply source/index.html.md:363]: we'll shore up the details later, I _think_ this is going to be a base64 field, but I'll confirm with Sergey tomorrow. Mojo need this on Dev today though

Comment (sonerayberk) [reply source/index.html.md:363]: `get-dip` API endpoint returns currently just a `decision` field

Comment (danyilmaz) [reply source/index.html.md:363]: yes, but that'll change.
Mojo aren't going to be using this yet, they're just going to use it to estimate their workload

**Review by sonerayberk — APPROVED** [review]

**Pull request merged**